spark/src/main/java/gaffer/spark/GafferTable.java

```diff
@@ -0,0 +1,333 @@
+     */
+    public JavaPairRDD<Element, Properties> query(final Collection<Object> vertices) throws RangeFactoryException {
+        Configuration conf = new Configuration();
+        config.setConfiguration(conf, vertices);
+        return query(conf, true);
+    }
+
+    /**
+     * Creates a {@link JavaPairRDD} by querying the table for a single
+     * {@link Object} vertex.
+     *
+     * @param vertex
+     *            The {@link Object} to be queried for.
+     * @return A {@link JavaPairRDD} of {@link Element}, {@link Properties}
+     *         pairs.
+     * @throws RangeFactoryException When an error occurs creating a range.
+     */
+    public JavaPairRDD<Element, Properties> query(final Object vertex) throws RangeFactoryException {
+        return query(Collections.singleton(vertex));
+    }
+
+    /**
+     * Creates a {@link JavaPairRDD} by querying the table for a
+     * {@link Collection} of ranges from {@link Pair}s of vertices.
+     *
+     * @param ranges
+     *            The {@link Pair}s of vertices to be queried for.
+     * @return A {@link JavaPairRDD} of {@link Element}, {@link Properties}
+     *         pairs.
+     * @throws RangeFactoryException When an error occurs creating a range.
+     */
+    public JavaPairRDD<Element, Properties> rangeQuery(final Collection<Pair<Object>> ranges) throws RangeFactoryException {
+        Configuration conf = new Configuration();
+        config.setConfigurationFromRanges(conf, ranges);
+        return query(conf, true);
+    }
+
+    /**
+     * Creates a {@link JavaPairRDD} by querying the table for a range from a
+     * {@link Pair} of vertices.
+     *
+     * @param range
+     *            The {@link Pair} of vertices to be queried for.
+     * @return a {@link JavaPairRDD} of {@link Element}, {@link Properties}
+     *         pairs.
+     * @throws RangeFactoryException When an error occurs creating a range.
+     */
+    public JavaPairRDD<Element, Properties> rangeQuery(final Pair<Object> range) throws RangeFactoryException {
+        return rangeQuery(Collections.singleton(range));
+    }
+
+    /**
+     * Creates a {@link JavaPairRDD} by querying the table for an
+     * {@link Collection} of pairs of {@link Object} vertices.
+     *
+     * Only those edges between the two entities will be returned.
+     *
+     * @param pairs
+     *            The pairs of {@link Object} vertices to be queried for.
+     * @return a {@link JavaPairRDD} of {@link Element}, {@link Properties}
+     *         pairs.
+     * @throws RangeFactoryException When an error occurs creating a range.
+     */
+    public JavaPairRDD<Element, Properties> pairQuery(final Collection<Tuple2<Object, Object>> pairs) throws RangeFactoryException {
+        Configuration conf = new Configuration();
+
+        Collection<Pair<Object>> coll = new HashSet<>();
+        for (Tuple2<Object, Object> t : pairs) {
+            coll.add(new Pair<>(t._1, t._2));
```

Comment:
Although the code does compile, I am getting errors in my IDE. We should probably be using accessor methods for these variables instead. 

e.g:
```java
new Pair<>(t._1(), t._2())
```